instrumentation/kamon-spring/src/main/scala/kamon/instrumentation/spring/server/SpringMVCInstrumentation.scala

```diff
@@ -0,0 +1,95 @@
+package kamon.instrumentation.spring.server
+
+import kamon.Kamon
+import kamon.context.Storage.Scope
+import kamon.instrumentation.http.HttpServerInstrumentation.RequestHandler
+import kamon.trace.Span
+import kanela.agent.api.instrumentation.InstrumentationBuilder
+import kanela.agent.libs.net.bytebuddy.asm.Advice
+import kanela.agent.libs.net.bytebuddy.matcher.ElementMatchers.{named, takesArgument}
+import org.springframework.web.servlet.HandlerMapping
+
+import java.util.concurrent.Callable
+import javax.servlet.http.{HttpServletRequest, HttpServletResponse}
+
+class SpringMVCInstrumentation extends InstrumentationBuilder {
+
+  /*
+   * Adds serverInstrumentation to DispatcherServlet, measures execution time
+   * of calling the doDispatch method and propagates context
+   */
+  onType("org.springframework.web.servlet.DispatcherServlet")
+    .mixin(classOf[HasServerInstrumentation.Mixin])
+    .advise(method("doDispatch"), DispatchAdvice)
+    .advise(method("render"), RenderAdvice)
+    .advise(method("processHandlerException"), ProcessHandlerAdvice)
+
+  /*
+   * Changes Callable argument of startCallableProcesing with an
+   * instrumented one that stores the context when called.
+   */
+
+  onType("org.springframework.web.context.request.async.WebAsyncManager")
+    .advise(
+      method("startCallableProcessing")
+        // is this [_] ok?
+        .and(withArgument(0, classOf[Callable[_]])), classOf[CallableWrapper])
+}
+
+object ProcessHandlerAdvice {
+  @Advice.OnMethodExit(onThrowable = classOf[Throwable], suppress = classOf[Throwable])
+  def exit(@Advice.Argument(3) throwable: Throwable): Unit =
+    Option(throwable).map(Kamon.currentSpan().fail)
```

Comment:
Are you sure that when a Span fails it will still be finished somewhere else in the instrumentation?

Reply:
Yes!

Reply:
I think that is good idea, when were are in the hot path, make an explicit null check instead of using an Option in order to avoid an unnecessary allocation. :)